File: BloodInNeed/Models/BloodRequest.cs
using BloodInNeed.UI.Controllers;

namespace BloodInNeed.UI.Models
{
    public class BloodRequest
    {
        
        public int receipentId {get; set;}
        public string patientName {get; set;}
        public int bloodGroup {get; set;}
        public string bloodGroupSymbol {get; set;}
        public int requiredUnits {get; set;}
        public string urgencyLevel {get; set;}
        public string hospitalName {get; set;}
        public int bloodCity {get; set;}
        public DateTime dueDate {get; set;}
        public string contactNumber {get; set;}
        public string diagnosis {get; set;}
        public string notes {get; set;}
      

    }



    public class MyRequests
    {
        public int SN { get; set; }
        public int RequestId { get; set; }
        public int receipentId { get; set; }
        public string receipentName { get; set; }
        public string patientName { get; set; }
        public string bloodGroup { get; set; }
        public double requiredUnits { get; set; }
        public string urgencyLevel { get; set; }
        public string hospitalName { get; set; }
        public int bloodCity { get; set; }
        public string CityName { get; set; }
        public DateTime dueDate { get; set; }
        public string contactNumber { get; set; }
        public string diagnosis { get; set; }
        public string notes { get; set; }
        public string Status { get; set; }
        public DateTime RequestedOn { get; set; }
        public string AcceptedOn { get; set; }
        public DateTime CancelledOn { get; set; }
        public string Donor { get; set; }
        public string CancelledBy { get; set; }

        public string ReceipentContact { get; set; }
        public string DonorContact { get; set; }



    }

    public class ViewRequests
    {
        public int SN { get; set; }
        public int RequestId { get; set; }
        public string ReceipientName { get; set; }
        public string PatientName { get; set; }
        public string bloodGroup { get; set; }
        public double requiredUnits { get; set; }

        public string urgencyLevel { get; set; }
        public string hospitalName { get; set; }
        public int bloodCity { get; set; }
        public string CityName { get; set; }
        public DateTime dueDate { get; set; }
        public string contactNumber { get; set; }
        public string diagnosis { get; set; }
        public string notes { get; set; }
        public string Status { get; set; }
        public DateTime RequestedOn { get; set; }
        public DateTime AcceptedOn { get; set; }

        public int IsEligibleToDonate { get; set; }
        public string ReceipentContact { get; set; }
        public string DonorContact { get; set; }

    }


}

File: BloodInNeed/wwwroot/js/BloodRequest.js

var apiUrl = '/api/BloodRequest/';

var BloodRequest = {

     msg : '',
     msgType : '',
    init: function () {
               

        $('#saveBtn').on('click', function () {
            
            showProcessing();

            BloodRequest.validateForm();

            if (msgType == "error") {
                showToast(msg, msgType);
                $('#errorMsg').text(msg);
                hideProcessing();
                return;
            }
            if (msgType == "success") {

                var data =
                {
                    
                    receipentId : $('#currentLoggedInUserId').val(),
                    patientName : $('#patientName').val(),
                    bloodGroup : $('#bloodGroup option:selected').val(),
                    requiredUnits : $('#requiredUnits').val(),
                    urgencyLevel : $('#urgencyLevel option:selected').val(),
                    hospitalName : $('#hospitalName').val(),
                    bloodCity : $('#bloodCity option:selected').val(),
                    dueDate : $('#dueDate').val(),
                    contactNumber : $('#contactNumber').val(),
                    diagnosis : $('#diagnosis').val(),
                    notes: $('#notes').val(),
                    bloodGroupSymbol : ''
                    
                }


                ApiCall.post(apiUrl + 'CreateRequest', data, function (result) {
                    BloodRequest.CreateRequestnCallBack(result);
                }, BloodRequest.failureCallback());

            }

            
        });


        $('.view-request').on('click', function () {
            
            var requestId = $(this).data("id");

            // Show modal immediately with loader
            $("#requestDetailsContent").html(`
            <div class="text-center py-4">
                <div class="spinner-border text-primary" role="status">
                    <span class="visually-hidden">Loading...</span>
                </div>
            </div>
        `);
            $("#requestDetailsModal").modal("show");


        });




    }

    , validateForm: function () {

        var receipentId = $('#currentLoggedInUserId').val();
        var patientName = $('#patientName').val();
        var bloodGroup = $('#bloodGroup option:selected').val();
        var requiredUnits = $('#requiredUnits').val();
        var urgencyLevel = $('#urgencyLevel option:selected').val();
        var hospitalName = $('#hospitalName').val();
        var bloodCity = $('#bloodCity option:selected').val();
        var dueDate = $('#dueDate').val();
        var contactNumber = $('#contactNumber').val();
        var diagnosis = $('#diagnosis').val();
        var notes = $('#notes').val();

        if (
            patientName == '' ||
            bloodGroup == '' ||
            requiredUnits == '' || requiredUnits == 0 ||
            urgencyLevel == '' ||
            hospitalName == '' ||
            bloodCity == '' || bloodCity == '' ||
            dueDate == 0 || dueDate == '' || dueDate == null ||
            dueDate == 0 || dueDate == '' || dueDate == null ||
            diagnosis == ''
        ) {
            msg = "Fill all essential details.";
            msgType = "error";
        }
        else {
            msg = "All essential field filled.";
            msgType = "success";
        }

    }

    , CreateRequestnCallBack: function (result) {
        showProcessing();
        //console.log(result);
        
        // Show toast notification based on the result
        if (result.msgType === "success") {

            showProcessing();


            setTimeout(() => {
                window.location.href = "/BloodSearch/MyRequests";

                showToast(result.msg, "success"); // Display success toast
                hideProcessing();
            }, 3000);
        }

            else {
            showToast(result.msg, "error"); // Display error toast
        }

        hideProcessing();

    }




    , failureCallback: function () {

        console.log("Api Call Failed.");
        
    }


 }

BloodRequest.init();


